quarkchain/cluster/shard.py

```diff
@@ -297,14 +301,18 @@ def __init__(self, header: MinorBlockHeader, shard_conn: PeerShardConnection):
         while len(block_header_chain) > 0:
             block_chain = await asyncio.wait_for(
                 self.__download_blocks(block_header_chain[:MINOR_BLOCK_BATCH_SIZE]),
-                TIMEOUT,
+                SYNC_TIMEOUT,
             )
             Logger.info(
                 "[{}] downloaded {} blocks from peer".format(
                     self.shard_state.branch.to_str(), len(block_chain)
                 )
             )
-            check(len(block_chain) == len(block_header_chain[:MINOR_BLOCK_BATCH_SIZE]))
+            if len(block_chain) != len(block_header_chain[:MINOR_BLOCK_BATCH_SIZE]):
+                # TODO: tag bad peer
+                return self.shard_conn.close_with_error(
```

Comment:
I think the protocol should be
- the peer could response a smaller amount of blocks since it may implement its local response limit
- The responded blocks should maintain the same order as requested
- The responded blocks should be at least 1

Reply:
I think there is freedom in how this is handled. The remote peer is free to respond with any number of blocks they may want to return (this PR fixes the potential risk that an evil peer will crash this node) and this node is free to disconnect if the number does not meet expectations